src/components/button-bar/button-bar.test.js

```diff
@@ -77,4 +81,65 @@ context("Test for Button-Bar component", () => {
       cy.checkAccessibility();
     });
   });
+
+  describe("when custom Button wrapper components are used as children in ButtonBar", () => {
+    const WrappedComponent = () => {
+      return (
+        <>
+          <Button iconType="bin">bar</Button>
+          <Button iconType="csv">bar</Button>
+          <Button iconType="pdf">bar</Button>
+        </>
+      );
+    };
+
+    it("Button size is small / 81px when the size prop is set to small and passed to ButtonBar", () => {
+      CypressMountWithProviders(
+        <ButtonBar size="small">
+          <WrappedComponent />
+          <IconButton onAction={() => undefined}>
+            <Icon type="csv" />
+          </IconButton>
+        </ButtonBar>
+      );
+
+      buttonDataComponent().then(($el) => {
+        useJQueryCssValueAndAssert($el, "width", 81);
+      });
+    });
+
+    it("Button is fullWidth / 339px when the fullWidth prop is passed to ButtonBar", () => {
+      CypressMountWithProviders(
+        <ButtonBar fullWidth>
+          <WrappedComponent />
+          <IconButton onAction={() => undefined}>
+            <Icon type="csv" />
+          </IconButton>
+        </ButtonBar>
+      );
+
+      buttonDataComponent().then(($el) => {
+        useJQueryCssValueAndAssert($el, "width", 339);
+      });
+    });
+
+    it.each([
```

Comment:
It would be good to test focus and keyboard nav works with wrapped children. I also think for the other props (from context) you can just replicate the existing ones but pass in your WrappedChildren (or whatever you call it) to the `CypressMountWithProviders`